CONTRIBUTING.md

```diff
@@ -192,3 +213,72 @@ Go to the Actions tab and click the "Drop Old Version" tab. Then click the "Run
+
+There are probably many other footguns at the moment, feel free to update this with any you can think of!
+
+## How IaSQL PostgreSQL functions work
+
+IaSQL [functions or operations](https://docs.iasql.com/function) like `iasql_install(..)`, `iasql_sync()`, `iasql_apply()` are required to provide a synchronous user experience within a PostgreSQL connection such that the user can't accidentally alter data in tables while an `apply` is taking place whilst still implementing the majority of the logic behind the operations in Node.js. It is worth pointing out that fulfilling this requirement doesn't stop someone from opening two simultaneous connections to the same database or creating two databases pointing to the same cloud account.
+
+[Graphile worker](https://github.com/graphile/worker) is a job queue that runs on PostgreSQL + Node.js. The engine has one Graphile worker per database. Graphile worker setup and job definition is defined in the [scheduler.ts](https://github.com/iasql/iasql-engine/blob/main/src/services/scheduler.ts). At startup, the Node.js engine looks at its [metadata repository](https://github.com/iasql/iasql-engine/blob/main/src/entity/index.ts) which stores information about all the IaSQL databases the engine is managing so it can spin up a Graphile worker per existing IaSQL database and put it into it's own Node.js child process to avoid clogging up the event loop. New databases created via the `/connect` route get a new metadata repo entry and a new Graphile worker within a child process.
+
+Graphile workers are used to implement an [RPC](https://github.com/iasql/iasql-engine/blob/main/src/services/scheduler-api.ts) in which the IaSQL functions in SQL can call into the actual logic of operations that live in the Node.js engine. The PG SQL functions poll until the Node.js code is done via the [`iasql_operation`](https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.18/iasql_functions/entity/index.ts) table which is used to keep the output, the state, and the finish time for every IaSQL operation (`apply`, `sync`, `install`, etc.) in that database. The `until_iasql_operation` PG SQL function is ultimately called by every IaSQL function and is what kicks off the polling. This function invokes the Node.js code via the database's Graphile worker by using `graphile_worker.add_job(...)` and then creates a while loop watching for the `end_time` column in `iasql_operation` table to wait for the result of the Node.js code.
```

Comment:
Technically this seems to be an [IPC](https://stackoverflow.com/a/2161729) instead of RPC, since both the processes are on the same machine.

Reply:
Let's also make "Graphile worker" hyperlinked to `https://github.com/graphile/worker`